src/main/java/it/pagopa/ecommerce/commons/utils/NpgApiKeyHandler.java

```diff
@@ -0,0 +1,173 @@
+package it.pagopa.ecommerce.commons.utils;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.vavr.control.Either;
+import it.pagopa.ecommerce.commons.client.NpgClient;
+import it.pagopa.ecommerce.commons.exceptions.NpgApiKeyConfigurationException;
+
+import java.util.*;
+import java.util.function.Function;
+
+/**
+ * This class handles all NPG api keys such as default one and per methods psp.
+ * Use {@link NpgApiKeyHandlerBuilder} class to initialize a new class instance
+ * api keys
+ */
+public class NpgApiKeyHandler {
+
+    /**
+     * NpgApiKeyHandler builder class
+     */
+    public static class NpgApiKeyHandlerBuilder {
+        private String defaultApiKey;
+        private final Map<NpgClient.PaymentMethod, NpgPspApiKeysConfig> methodsApiKeyMapping = new EnumMap<>(
+                NpgClient.PaymentMethod.class
+        );
+
+        /**
+         * Set the default NPG api key
+         *
+         * @param defaultApiKey the default api key to be set
+         * @return this builder instance
+         */
+        public NpgApiKeyHandlerBuilder setDefaultApiKey(String defaultApiKey) {
+            this.defaultApiKey = defaultApiKey;
+            return this;
+        }
+
+        /**
+         * Add specific {@link NpgPspApiKeysConfig} configuration for input payment
+         * method
+         *
+         * @param paymentMethod       the payment method for which register npg api keys
+         *                            configuration
+         * @param npgPspApiKeysConfig the npg api keys config
+         * @return this builder instance
+         */
+        public NpgApiKeyHandlerBuilder addMethodPspMapping(
+                                                           NpgClient.PaymentMethod paymentMethod,
+                                                           NpgPspApiKeysConfig npgPspApiKeysConfig
+        ) {
+            if (this.methodsApiKeyMapping.containsKey(paymentMethod)) {
+                throw new NpgApiKeyConfigurationException(
+                        "Api key mapping already registered for payment method: [%s]".formatted(paymentMethod)
+                );
+            }
+            this.methodsApiKeyMapping.put(paymentMethod, npgPspApiKeysConfig);
+            return this;
+        }
+
+        /**
+         * Add specific {@link NpgPspApiKeysConfig} configuration for input payment
+         * method. This method is the same as
+         * {@link #addMethodPspMapping(NpgClient.PaymentMethod, NpgPspApiKeysConfig)}
+         * one but parse psp api keys configuration with the given input information
+         *
+         * @param paymentMethod the payment methods to be added to mapping
+         * @param pspKeys       the required psp keys
+         * @param pspToHandle   the psp expected to be handled
+         * @param objectMapper  the object mapper instance to be used to parse api keys
+         *                      configuration
+         * @return this builder instance
+         */
```

Comment:
[docs] I'd add either:
 * that the parsing semantics depend on `NpgPspApiKeysConfig#parseApiKeyConfiguration`, or
 * that the expected string format is JSON
wdyt?

Reply:
Also `@throws` :)

Reply:
Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/188/commits/17f5fabeeddc26d002749691174e3a9fde598cf2